lib/platform/azure/azure-helper.ts

```diff
@@ -5,12 +5,35 @@ import {
   GitPullRequest,
 } from 'azure-devops-node-api/interfaces/GitInterfaces';
 
+import { Options } from 'simple-git/promise';
 import * as azureApi from './azure-got-wrapper';
 import { logger } from '../../logger';
 import { Pr } from '../common';
+import { HostRule } from '../../util/host-rules';
 
 const mergePolicyGuid = 'fa4e907d-c16b-4a4c-9dfa-4916e5d171ab'; // Magic GUID for merge strategy policy configurations
 
+export function getStorageExtraCloneOpts(config: HostRule): Options {
+  let header: string;
+  const headerName = 'AUTHORIZATION';
+  switch (config.authenticationType) {
+    case 'basic':
+      header = `${headerName}: basic ${Buffer.from(
+        `${config.username}:${config.password}`
+      ).toString('base64')}`;
+      break;
+    case 'bearerToken':
+      header = `${headerName}: bearer ${config.token}`;
+      break;
+    case 'personalAccessToken':
+    default:
+      header = `${headerName}: basic ${Buffer.from(`:${config.token}`).toString(
```

Comment:
This can be unified with username / password, as this is the same as using an empty username and the token as password.

So we only have username / password and token auth as before and don't need `authenticationType`

Reply:
I see what you mean, but:
* That would make it a breaking change. For example the [azure extension](https://marketplace.visualstudio.com/items?itemName=jyc.vsts-extensions-renovate-me) would need to be changed. Maybe an option is to use the same logic as [getHandlerFromToken](https://github.com/microsoft/azure-devops-node-api/blob/417f07133a7f6e2b18745afbdca6268fcc7801a8/api/WebApi.ts#L66) instead? Checking on token length seems a bit dirty, but well if the azure api does it internally, I guess it can also be used here.
* Azure has another auth type [here](https://github.com/microsoft/azure-devops-node-api/blob/417f07133a7f6e2b18745afbdca6268fcc7801a8/api/WebApi.ts#L54), which I didn't include because I am not familiar with it and had no idea how to test. Maybe if in the future someone is interested in it, it could be easier to add if authenticationType is there?

Reply:
pending discussion, I've committed an update [here](https://github.com/kroonprins/renovate/commit/f8338ce5a9e9b640adf49466623f48ffae27b0f6), if this is preferred I can push that to this PR

Reply:
Can you pull that commit into this PR? I think it simplifies this by not requiring an extra config option. One thing though: I think it might be required to have a non-empty username when using a PAT. Currently we use `token:<PAT>`.

Reply:
done

Reply:
About `token:<PAT>`: I tested it with `:<PAT>` and it worked on my side

Reply:
Thanks for testing.